src/server/game/Spells/Spell.cpp

```diff
@@ -2295,6 +2295,10 @@ void Spell::DoAllEffectOnTarget(TargetInfo* target)
                 m_caster->ToCreature()->LowerPlayerDamageReq(target->damage);
         }
     }
+    else if (missInfo == SPELL_MISS_IMMUNE2)
+    {
+        m_caster->CombatStart(ObjectAccessor::GetUnit(*m_caster, target->targetGUID)); // Initiate combat with target, despite target being immune
```

Comment:
m_caster->CombatStart(unit);

Reply:
I would suggest using unit instead of "ObjectAccessor::GetUnit(*m_caster, target->targetGUID)", or at least I don't see a reason not to use it.
I wonder tho if it's correct to skip the "if (spellHitTarget)" check below with its pvp related logic